src/app/productConfig/productConfigModal/productConfig.modal.component.js

```diff
@@ -35,69 +20,57 @@ export const giveGiftParams = {
 class ProductConfigModalController {
 
   /* @ngInject */
-  constructor( $window, $location, $scope, $log, designationsService, cartService, modalStateService, analyticsFactory ) {
+  constructor( $window, $location, modalStateService ) {
     this.$window = $window;
     this.$location = $location;
-    this.$scope = $scope;
-    this.$log = $log;
-    this.designationsService = designationsService;
-    this.cartService = cartService;
     this.modalStateService = modalStateService;
-    this.possibleTransactionDays = possibleTransactionDays;
-    this.startDate = startDate;
-    this.analyticsFactory = analyticsFactory;
-
-    this.selectableAmounts = [50, 100, 250, 500, 1000, 5000];
   }
 
   $onInit(){
     this.initModalData();
     this.initializeParams();
-    this.setDefaultAmount();
-    this.waitForFormInitialization();
+  }
+
+  $onDestroy(){
+    // Clear the modal name and params when the modal closes
+    this.modalStateService.name( null );
+    angular.forEach( giveGiftParams, ( value ) => {
+      // Remove all query params except CampaignCode
+      if( value !== giveGiftParams.campaignCode ) {
+        this.$location.search(value, null);
+      }
+    } );
   }
 
   initModalData() {
-    this.productData = this.resolve.productData;
+    this.code = this.resolve.code;
     this.itemConfig = this.resolve.itemConfig;
     this.isEdit = this.resolve.isEdit;
     this.uri =  this.resolve.uri;
-    this.suggestedAmounts =  this.resolve.suggestedAmounts;
-    this.nextDrawDate = this.resolve.nextDrawDate;
-
-    if(!this.itemConfig['recurring-day-of-month'] && this.nextDrawDate) {
-      this.itemConfig['recurring-day-of-month'] = startDate(null, this.nextDrawDate).format('DD');
-    }
-
-    this.showRecipientComments = !!this.itemConfig['recipient-comments'];
-    this.showDSComments = !!this.itemConfig['donation-services-comments'];
-
-    this.useSuggestedAmounts = !isEmpty(this.suggestedAmounts);
   }
 
   initializeParams() {
     if(this.isEdit){
       return;
     }
 
-    let params = this.$location.search();
-
+    // Add query params for current modal
     this.modalStateService.name( 'give-gift' );
-    this.$location.search( giveGiftParams.designation, this.productData.code );
-    this.$location.search( giveGiftParams.campaignPage, this.itemConfig['campaign-page'] );
+    this.updateQueryParam( giveGiftParams.designation, this.code ); // TODO: can this.code change? Do we need to set this again if productData comes back with a different code?
```

Comment:
K. I was just making sure there wasn't some specific reason it wasn't using `this.code` in the first place.